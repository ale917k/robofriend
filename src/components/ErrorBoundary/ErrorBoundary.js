import React, { Component } from "react";

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = { hasErrored: false };
  }

  componentDidCatch(error, info) {
    this.setState({ hasErrored: true });
  }

  render() {
    if (this.state.hasErrored) {
      return <h1>Something went wrong.</h1>;
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
