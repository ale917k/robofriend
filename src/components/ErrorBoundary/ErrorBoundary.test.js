import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import React from "react";
import ErrorBoundary from "./ErrorBoundary";

describe("ErrorBoundary", () => {
  it("expects to match Snapshot Object", () => {
    const wrapper = shallow(<ErrorBoundary />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("errors if wrapped component throws error", () => {
    const Something = () => null;
    const wrapper = mount(
      <ErrorBoundary>
        <Something />
      </ErrorBoundary>
    );
    const error = new Error("test");

    expect(wrapper.state()).toEqual({ hasErrored: false });
    wrapper.find(Something).simulateError(error);
    expect(wrapper.state()).toEqual({ hasErrored: true });
  });
});
