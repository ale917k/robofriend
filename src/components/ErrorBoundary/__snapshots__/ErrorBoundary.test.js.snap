// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary expects to match Snapshot Object 1`] = `""`;
